rusk/src/bin/config/kadcast.rs

```diff
@@ -19,32 +19,34 @@ impl From<KadcastConfig> for Config {
 
 impl KadcastConfig {
     pub(crate) fn merge(&mut self, matches: &ArgMatches) {
-        if let Some(public_address) = matches.value_of("kadcast_public_address")
+        if let Some(public_address) =
+            matches.get_one::<String>("kadcast_public_address")
         {
             self.0.public_address = public_address.into();
         };
-        if let Some(listen_address) = matches.value_of("kadcast_listen_address")
-        {
-            self.0.listen_address = Some(listen_address.into());
-        };
+
+        self.0.listen_address =
```

Comment:
Same as previous comments but with an additional comment.

The `merge` method merge a toml configuration with the cli args.
If the listen_address is specified through the toml and not specified as cli arg, the result will be None

Reply:
If the `listen_address` is specified in the config.toml but no `listen_address` is explicitly set using the cli arg, then the listen_address will be None